### PR TITLE
feat(ext.pages): Add `use_styled_buttons`

### PR DESCRIPTION
**Pull request by Nzii3**
## Summary

Adds ``use_styled_buttons`` in `ext.pages.Paginator` which essentially styles buttons with emojis by using a single parameter. Overwrites ``use_default_buttons`` when used in the Paginator `__init__`. 

## Information

- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting,
      examples, ...).

## Checklist

- [x] I have searched the open pull requests for duplicates.
- [x] If code changes were made then they have been tested.
  - [x] I have updated the documentation to reflect the changes.
- [x] If `type: ignore` comments were used, a comment is also left explaining why.

### CODE REVIEW

**Comment by Nzii3** [comment]
Image of styled buttons:
image

**Comment by Nzii3** [comment]
Example code:
```py
from discord.ext.pages import Paginator
pages = ['1', '2', '3']
paginator = Paginator(pages=pages, use_styled_buttons=True)
await paginator.respond(ctx)
```
Output: 
image

**Comment by Lulalaby** [comment]
I like how the ci fails again

**Comment by Nzii3** [comment]
> I like how the ci fails again
@Lulalaby *it doesn’t like me :/*

File: discord/ext/pages/pagination.py [hunk 1]
```diff
@@ -254,6 +254,8 @@ class PageGroup:
         Whether the buttons get disabled when the paginator view times out.
     use_default_buttons: :class:`bool`
         Whether to use the default buttons (i.e. ``first``, ``prev``, ``page_indicator``, ``next``, ``last``)
+    use_styled_buttons: :class:`bool`
+        The same as ``use_default_buttons`` but uses emojis for labels, overwrites ``use_default_buttons`` when used
```
Comment (Lulalaby) [inline discord/ext/pages/pagination.py:258]:
```suggestion
        The same as ``use_default_buttons`` but uses emojis for labels, overwrites ``use_default_buttons`` when used.
```

File: discord/ext/pages/pagination.py [hunk 2]
```diff
@@ -489,11 +504,13 @@ async def update(
...
         menu_placeholder: :class:`str`
             The placeholder text to show in the page group menu when no page group has been selected yet.
-            Defaults to "Select Page Group" if not provided.
+            Defaults to "Select page group..." if not provided.
         disable_on_timeout: :class:`bool`
             Whether the buttons get disabled when the paginator view times out.
         use_default_buttons: :class:`bool`
             Whether to use the default buttons (i.e. ``first``, ``prev``, ``page_indicator``, ``next``, ``last``)
+        use_styled_buttons: :class:`bool`
+            The same as ``use_default_buttons`` but uses emojis for labels, overwrites ``use_default_buttons`` when used
```
Comment (Lulalaby) [inline discord/ext/pages/pagination.py:513]:
```suggestion
            The same as ``use_default_buttons`` but uses emojis for labels, overwrites ``use_default_buttons`` when used.
```

File: discord/ext/pages/pagination.py [hunk 3]
```diff
@@ -328,13 +332,15 @@ class Paginator(discord.ui.View):
...
-        Defaults to "Select Page Group" if not provided.
+        Defaults to "Select page group..." if not provided.
     author_check: :class:`bool`
         Whether only the original user of the command can change pages.
     disable_on_timeout: :class:`bool`
         Whether the buttons get disabled when the paginator view times out.
     use_default_buttons: :class:`bool`
         Whether to use the default buttons (i.e. ``first``, ``prev``, ``page_indicator``, ``next``, ``last``)
+    use_styled_buttons: :class:`bool`
+        The same as ``use_default_buttons`` but uses emojis for labels, overwrites ``use_default_buttons`` when used
```
Comment (Lulalaby) [inline discord/ext/pages/pagination.py:343]:
```suggestion
        The same as ``use_default_buttons`` but uses emojis for labels, overwrites ``use_default_buttons`` when used.
```

**Review by Lulalaby — CHANGES_REQUESTED** [review]

**Comment by Lulalaby** [comment]
No worries. Is fixed. Please address the requested changes.

**Comment by Nzii3** [comment]
Added changes 👍

**Review by Lulalaby — APPROVED** [review]

**Comment by Lulalaby** [comment]
I don't even question anymore how you've managed to let the ci fail two more times just by applying the string changes

**Comment by BobDotCom** [comment]
> I don't even question anymore how you've managed to let the ci fail two more times just by applying the string changes

CI sometimes fails when a new build is triggered while the previous isn't finished

File: discord/ext/pages/pagination.py [hunk 4]
```diff
@@ -254,6 +254,8 @@ class PageGroup:
         Whether the buttons get disabled when the paginator view times out.
     use_default_buttons: :class:`bool`
         Whether to use the default buttons (i.e. ``first``, ``prev``, ``page_indicator``, ``next``, ``last``)
+    use_styled_buttons: :class:`bool`
+        The same as ``use_default_buttons`` but uses emojis for labels, overwrites ``use_default_buttons`` when used.
```
Comment (BobDotCom) [inline discord/ext/pages/pagination.py:258]: Please add a versionadded for v2.4

File: discord/ext/pages/pagination.py [hunk 5]
```diff
@@ -328,13 +332,15 @@ class Paginator(discord.ui.View):
...
-        Defaults to "Select Page Group" if not provided.
+        Defaults to "Select page group..." if not provided.
     author_check: :class:`bool`
         Whether only the original user of the command can change pages.
     disable_on_timeout: :class:`bool`
         Whether the buttons get disabled when the paginator view times out.
     use_default_buttons: :class:`bool`
         Whether to use the default buttons (i.e. ``first``, ``prev``, ``page_indicator``, ``next``, ``last``)
+    use_styled_buttons: :class:`bool`
+        The same as ``use_default_buttons`` but uses emojis for labels, overwrites ``use_default_buttons`` when used.
```
Comment (BobDotCom) [inline discord/ext/pages/pagination.py:343]: Please add a versionadded for v2.4

File: discord/ext/pages/pagination.py [hunk 6]
```diff
@@ -489,11 +504,13 @@ async def update(
...
         menu_placeholder: :class:`str`
             The placeholder text to show in the page group menu when no page group has been selected yet.
-            Defaults to "Select Page Group" if not provided.
+            Defaults to "Select page group..." if not provided.
         disable_on_timeout: :class:`bool`
             Whether the buttons get disabled when the paginator view times out.
         use_default_buttons: :class:`bool`
             Whether to use the default buttons (i.e. ``first``, ``prev``, ``page_indicator``, ``next``, ``last``)
+        use_styled_buttons: :class:`bool`
+            The same as ``use_default_buttons`` but uses emojis for labels, overwrites ``use_default_buttons`` when used.
```
Comment (BobDotCom) [inline discord/ext/pages/pagination.py:513]: Please add a versionadded for v2.4

File: discord/ext/pages/pagination.py [hunk 7]
```diff
@@ -773,6 +795,45 @@ def add_default_buttons(self):
         for button in default_buttons:
             self.add_button(button)
 
+    def add_styled_buttons(self):
+        """Adds the default paginator buttons like ``.add_default_Buttons`` but uses emojis for labels"""
```
Comment (BobDotCom) [inline discord/ext/pages/pagination.py:799]: Please add a versionadded for v2.4

**Review by BobDotCom — CHANGES_REQUESTED** [review]
In addition to the other comments, please add a changelog entry for both the new params and new method.

File: discord/ext/pages/pagination.py (hunk 4 above)
Comment (BobDotCom) [reply discord/ext/pages/pagination.py:258]: Here's an example from another open PR https://github.com/Pycord-Development/pycord/pull/1809/files#diff-f10a1b29b4a4c6c39210d6566a9c849b92f0e7e35746c78c61be6ac8fcb5c417R184-R189

Comment (Nzii3) [reply discord/ext/pages/pagination.py:258]: Ok. should I have it as added in 2.4?

**Comment by Nzii3** [comment]
> In addition to the other comments, please add a changelog entry for both the new params and new method.

Done.

File: discord/ext/pages/pagination.py [hunk 8]
```diff
@@ -254,6 +254,8 @@ class PageGroup:
         Whether the buttons get disabled when the paginator view times out.
     use_default_buttons: :class:`bool`
         Whether to use the default buttons (i.e. ``first``, ``prev``, ``page_indicator``, ``next``, ``last``)
+    use_styled_buttons: :class:`bool` .. versionadded:: 2.4
+        The same as ``use_default_buttons`` but uses emojis for labels, overwrites ``use_default_buttons`` when used.
```
Comment (BobDotCom) [inline discord/ext/pages/pagination.py:258]: It should look something more like this :)
```suggestion
    use_styled_buttons: :class:`bool`
        The same as ``use_default_buttons`` but uses emojis for labels, overwrites ``use_default_buttons`` when used.
        
        .. versionadded:: 2.4
```

Comment (Nzii3) [reply discord/ext/pages/pagination.py:258]: mb :/

File: discord/ext/pages/pagination.py [hunk 9]
```diff
@@ -773,6 +799,47 @@ def add_default_buttons(self):
...
             self.add_button(button)
 
+    def add_styled_buttons(self):
+        """Adds the default paginator buttons like ``.add_default_Buttons`` but uses emojis for labels
+        .. versionadded:: 2.4
+        """
+        default_buttons = [
+            PaginatorButton(
+                "first",
+                label="⏮",
```
Comment (BobDotCom) [inline discord/ext/pages/pagination.py:809]:
```suggestion
                emoji="⏮",
```

File: discord/ext/pages/pagination.py [hunk 10]
```diff
@@ -773,6 +799,47 @@ def add_default_buttons(self):
...
+                style=discord.ButtonStyle.blurple,
+                row=self.default_button_row,
+            ),
+            PaginatorButton(
+                "prev",
+                label="◀",
```
Comment (BobDotCom) [inline discord/ext/pages/pagination.py:815]:
```suggestion
                emoji="◀",
```

File: discord/ext/pages/pagination.py [hunk 11]
```diff
@@ -773,6 +799,47 @@ def add_default_buttons(self):
...
+            ),
+            PaginatorButton(
+                "page_indicator",
+                style=discord.ButtonStyle.gray,
+                disabled=True,
+                row=self.default_button_row,
+            ),
+            PaginatorButton(
+                "next",
+                label="▶",
```
Comment (BobDotCom) [inline discord/ext/pages/pagination.py:828]:
```suggestion
                emoji="▶",
```

File: discord/ext/pages/pagination.py [hunk 12]
```diff
@@ -773,6 +799,47 @@ def add_default_buttons(self):
...
+                style=discord.ButtonStyle.green,
+                loop_label="↩",
+                row=self.default_button_row,
+            ),
+            PaginatorButton(
+                "last",
+                label="⏭",
```
Comment (BobDotCom) [inline discord/ext/pages/pagination.py:835]:
```suggestion
                emoji="⏭",
```

**Review by BobDotCom — CHANGES_REQUESTED** [review]
After testing, I found that the style is highly inconsistent on MacOS. I've included examples from both before and after my suggested changes.

Before: System emoji is used for both the first and last buttons. The "next" button appears fine, but the "prev" button is oversized and the bottom of the character is clipped off.
Screen Shot 2022-12-03 at 12 56 38 PM

After: There is a blue outline on the emojis, but full parity is achieved between each emoji and between all operating systems as the discord emoji set is used.
Screen Shot 2022-12-03 at 1 00 25 PM

**Comment by Nzii3** [comment]
Should I add #1818 into this pr for v2.4?

**Comment by Lulalaby** [comment]
> Should I add #1818 into this pr for v2.4?

Personally I would prefer this

**Comment by Lulalaby** [comment]
@Nzii3 trash?

**Comment by Nzii3** [comment]
> @Nzii3 trash?

@Lulalaby yes

**Pull request closed without merging**